Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerBase.cpp

```diff
@@ -428,11 +429,15 @@ bool MediaPlayerPrivateGStreamerBase::handleSyncMessage(GstMessage* message)
         }
         GST_DEBUG("handling drm-preferred-decryption-system-id need context message");
         LockHolder lock(m_protectionMutex);
-        std::pair<Vector<GRefPtr<GstEvent>>, Vector<String>> streamEncryptionInformation = extractEventsAndSystemsFromMessage(message);
+        std::pair<Vector<GRefPtr<GstEvent>>, Vector<String>> streamEncryptionInformation = GStreamerEMEUtilities::extractEventsAndSystemsFromMessage(message);
         GST_TRACE("found %" G_GSIZE_FORMAT " protection events", streamEncryptionInformation.first.size());
         Vector<uint8_t> concatenatedInitDataChunks;
         unsigned concatenatedInitDataChunksNumber = 0;
         String eventKeySystemIdString;
+#if USE(OPENCDM)
+        HashMap<String, uint32_t> keySystemProtectionEventMap;
+        Vector<char> initDataChunks;
```

Comment:
We are just following the method how concatenated  initdata is working. Otherwise the initData session comparison will fail. As per your comment, we changed this to Vector<uint8_t> and used concatenatedInitDataChunks. But with each concatenatedInitDataChunks, there will be multiple events associated for WV, PR and ClearKey

Reply:
during RunLoop::main().dispatch, concatenatedInitDataChunks is copying using WTFMove, so the data will not be available after that, so we have used initDataChunks to keep initData value and to add into m_initDataProtectionEventsMapping once the handleSyncMessage detect the keySystem